src/src_/security.toit

```diff
@@ -1,32 +1,45 @@
+// Copyright (C) 2024 Toitware ApS.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the package's LICENSE file.
+
 import crypto.aes show AesGcm
+import io
 import protobuf
-import .utils
 import ..provision
 import ..proto_.session-pb
 import ..proto_.sec0-pb
 import ..proto_.sec2-pb
 import ..proto_.constants-pb
 import ..srp
 
-/** This security mode 0 doesn't encrypt/decrypt. */
-SECURITY0 := Security0_
+class SecurityCredentials:
+  salt/ByteArray
+  verifier/ByteArray
 
-/** This security mode 2 use SRP6a + AES-GCM. */
-security2 --salt/ByteArray --verifier/ByteArray -> Security:
-  return Security2_ salt verifier
+  constructor --.salt/ByteArray --.verifier/ByteArray:
 
-class Security0_ implements Security:
+security-for --credentials/SecurityCredentials? -> Security:
+  if not credentials: return Security0_
+  return Security2_ credentials.salt credentials.verifier
 
-  handshake data/ByteArray -> ByteArray:
-    resp-msg := null
+interface Security:
+  encrypt data/ByteArray -> ByteArray
+  decrypt data/ByteArray -> ByteArray
+  handle-handshake-request data/ByteArray -> protobuf.Message
+  version -> int
 
+/** This security mode 0 doesn't encrypt/decrypt. */
```

Comment:
Is there any other security mode 0? Maybe just `/** Security mode 0 ...`.

Reply:
done.